bufferserver/src/main/java/com/datatorrent/bufferserver/internal/PhysicalNode.java

```diff
@@ -71,36 +71,29 @@ public long getUptime()
    * @param d
    * @throws InterruptedException
    */
-  private SerializedData blocker;
-
   public boolean send(SerializedData d)
   {
-    if (d.offset == d.dataOffset) {
-      if (client.write(d.buffer, d.offset, d.length)) {
-        return true;
-      }
-    } else {
-      if (client.send(d.buffer, d.offset, d.length)) {
-        return true;
-      }
+    if (client.send(d.buffer, d.dataOffset, d.length - (d.dataOffset - d.offset))) {
```

Comment:
Looks like you are not sending the length. Is this correct? Also I notice that in earlier code there can be two scenarios. One where the caller has not populated length in the SerializedData and it is handled by the netlet client and one where the caller has and then it is a normal send by netlet. Won't we have two scenarios anymore?